scripts/spack/configs/docker/ubuntu22/spack.yaml

```diff
@@ -165,7 +165,7 @@ spack:
     scr:
       require: "@3.0.1~shared~tests~examples"
     umpire:
-      require: "@2024.07.0~shared~examples~werror"
+      require: "@2024.07.0+shared~examples~werror"
```

Comment:
@rhornung67 -- the change to the docker TPL configuration is actually in this line -- we are now forcing Umpire to build as a shared library for better compatibility w/ the shared axom config.